### PR TITLE
Fetching swagger url from assets API when Jazz developer clicks 'Test API' button

### PR DESCRIPTION
**Pull request by MichaelChinn**
Swagger file location found by parsing service assets

### CODE REVIEW

File: core/jazz-ui/src/app/pages/environment-detail/environment-detail.component.ts [hunk 1]
```diff
@@ -186,11 +182,41 @@ api_doc_name:string='';
...
   testApi(type){
       switch(type){
           case 'api':
-            let swaggerFile = '/' + this.service.domain + '/' + this.service.name + '/' + this.envSelected + '/swagger.json';
-            return window.open(environment.urls['swagger_editor'] + '/?url=' + environment['api_doc_name'] + swaggerFile);
+            let foundAsset = this.assets.find((asset) => {
+              return asset.asset_type === 'swagger_url';
+            });
+            if (foundAsset) {
+              return window.open(environment.urls['swagger_editor'] + '/?url=' + foundAsset.provider_id);
```
Comment (suryajak) [inline core/jazz-ui/src/app/pages/environment-detail/environment-detail.component.ts]: We don't want to hard code `?url=` so we have now included that as part of the key's value: https://github.com/tmobile/jazz-installer/pull/276/files#diff-04f485a5f936d3ceecd19be0e841d392R56.

Also we have slightly changed the key name to reflect its value, so reference it accordingly.

File: core/jazz-ui/src/environments/environment.oss.ts [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
   configFile: 'config/config.oss.json',
-  baseurl: "https://{API_GATEWAY_KEY_PROD}.execute-api.{inst_region}.amazonaws.com/prod",
-  api_doc_name: "https://{api_doc_name}.s3.amazonaws.com",
-  envName: "oss",
-  multi_env: {multi_env},
-  serviceTabs: ["{overview}", "{access control}", "{metrics}", "{logs}", "{cost}"],
-  environmentTabs: ["{env_overview}", "{deployments}", "{code quality}", "{assets}", "{env_logs}"],
-  urls: {
-    docs_link: "https://github.com/tmobile/jazz/wiki",
+  baseurl: "https://onbnj3emig.execute-api.us-east-1.amazonaws.com/prod",
```
Comment (suryajak) [inline core/jazz-ui/src/environments/environment.oss.ts]: Can you restore this?

File: core/jazz-ui/src/app/pages/environment-detail/environment-detail.component.ts [hunk 3]
```diff
@@ -151,53 +152,84 @@ export class EnvironmentDetailComponent implements OnInit {
...
+            } else {
+              return window.open('/404');
+            }
+          case 'website' :
+          if(this.endpoint_env!=(undefined||'')){
```
[the remark this answers is not in the document]
Comment (MichaelChinn) [reply core/jazz-ui/src/app/pages/environment-detail/environment-detail.component.ts]: Correct, I have updated

File: core/jazz-ui/src/app/pages/environment-detail/environment-detail.component.ts [hunk 4]
```diff
@@ -151,53 +152,84 @@ export class EnvironmentDetailComponent implements OnInit {
...
+            window.open(this.endpoint_env);
+          }
+          break;
+          case 'function' :
+          if(this.endpoint_env!=(undefined||'')){
+            window.open('/404');
+          }
+          break;
+          case 'lambda' :
+          this.setSidebar('try-service');
```
[the remark this answers is not in the document]
Comment (MichaelChinn) [reply core/jazz-ui/src/app/pages/environment-detail/environment-detail.component.ts]: handled

File: core/jazz-ui/src/environments/environment.oss.ts (hunk 2 above)
Comment (MichaelChinn) [reply core/jazz-ui/src/environments/environment.oss.ts]: handled

File: core/jazz-ui/Jenkinsfile_Platform [hunk 5]
```diff
@@ -4,10 +4,10 @@ import groovy.json.JsonSlurper
 import groovy.transform.Field
 
 // To be replaced as @Field def repo_credential_id = "value" except for config_loader. instance_prefix, jazz_bucket_web to be taken from the config_loader.
-@Field def repo_credential_id
-@Field def repo_base
-@Field def repo_core
-@Field def scm_type
+@Field def repo_credential_id = "jazz_repocreds"
```
Comment (suryajak) [inline core/jazz-ui/Jenkinsfile_Platform]: undo changes to lines 7 - 10

File: core/jazz-ui/src/app/pages/environment-detail/environment-detail.component.ts [hunk 6]
```diff
@@ -151,53 +152,80 @@ export class EnvironmentDetailComponent implements OnInit {
...
-        this.isLoadingService = false;
-        let errorMessage = this.messageservice.errorMessage(err, "serviceDetail");
-        this.toast_pop('error', 'Oops!', errorMessage)
+
+  fetchService(id: string){
+      this.isLoadingService = true;
+          this.subscription = this.http.get('/jazz/services/'+id).subscribe(
+            response => {
+              this.service.accounts=env_internal.urls.accounts;
+                    this.service.regions=env_internal.urls.regions;
```
Comment (suryajak) [inline core/jazz-ui/src/app/pages/environment-detail/environment-detail.component.ts]: indentation

File: core/jazz-ui/src/environments/environment.oss.ts [hunk 7]
```diff
@@ -2,13 +2,14 @@ export const environment = {
...
-  multi_env: {multi_env},
-  serviceTabs: ["{overview}", "{access control}", "{metrics}", "{logs}", "{cost}"],
-  environmentTabs: ["{env_overview}", "{deployments}", "{code quality}", "{assets}", "{env_logs}"],
-  urls: {
-    docs_link: "https://github.com/tmobile/jazz/wiki",
+  api_doc_name : "https://{api_doc_name}.s3.amazonaws.com",
+  envName:"oss",
+  multi_env:{multi_env},
+  serviceTabs:["{overview}","{access control}","{metrics}","{logs}","{cost}"],
+  environmentTabs:["{env_overview}","{deployments}","{code quality}","{assets}","{env_logs}"],
```
[the remark this answers is not in the document]
Comment (MichaelChinn) [reply core/jazz-ui/src/environments/environment.oss.ts]: That would be better but I do not know the full scope of how that variable is being used.

File: core/jazz-ui/src/app/pages/environment-detail/environment-detail.component.ts (hunk 6 above)
Comment (MichaelChinn) [reply core/jazz-ui/src/app/pages/environment-detail/environment-detail.component.ts]: indented

File: core/jazz-ui/Jenkinsfile_Platform (hunk 5 above)
Comment (MichaelChinn) [reply core/jazz-ui/Jenkinsfile_Platform]: reset

**Pull request merged**